packages/editor/src/package/storage-format.ts

```diff
@@ -0,0 +1,133 @@
+import * as t from 'io-ts'
+
+import { getEditorVersion } from './editor-version'
+
+/** The creator of the saved data -> Serlo editor */
+const documentType = 'https://serlo.org/editor'
+
+type Migration = (state: { version: number }) => { version: number }
+
+const migrations: Migration[] = [
+  // Migration: Add `editorVersion`
+  (state): StorageFormat => {
+    const expectedType = t.type({
+      type: t.literal(documentType),
+      variant: EditorVariantType,
+      version: t.number,
+      dateModified: t.string,
+      document: t.type({
+        plugin: t.string,
+        state: t.unknown,
+      }),
+    })
+    if (!expectedType.is(state))
+      throw new Error(
+        `Unexpected type during migration. Expected ${JSON.stringify(expectedType)} but got ${JSON.stringify(state)}`
+      )
+
+    const editorVersion = getEditorVersion()
+
+    return {
+      ...state,
+      editorVersion,
+    }
+  },
+  // ...
+  // Add new migrations here. Make sure the last one returns the new StorageFormat.
+]
+
+const currentVersion = migrations.length
+
+export function createEmptyDocument(
+  editorVariant: EditorVariant
+): StorageFormat {
+  return {
+    type: documentType,
+    variant: editorVariant,
+    version: currentVersion,
+    editorVersion: getEditorVersion(),
+    dateModified: getCurrentDatetime(),
+    document: {
+      plugin: 'type-generic-content',
+      state: {
+        content: {
+          plugin: 'rows',
+          state: [
+            {
+              plugin: 'text',
+              state: [
+                {
+                  type: 'p',
+                  children: [{ text: '' }],
+                },
+              ],
+            },
+          ],
```

Comment:
I think this might not be necessary.

Reply:
Ah, because it will be added anyway by the rows plugin at runtime? Hm, I actually prefer this to be more explicit and already a "valid" state from the beginning.

Reply:
What I'd personally prefer about something `{plugin: EditorPluginTypes.Rows}` instead of this:
- using the enum type like we do almost everywhere else
- leaving the correct initial/empty state of the plugin in the responsibility of the plugin
- less code

But I'm also okay with this way.

Reply:
Oh, sure. We should use the enum values. Did not see that until now. :see_no_evil: 

Hm, making the plugin responsible for the initial state sounds nice actually. You convinced me! Thank you for the explanation!

Reply:
Nice, let's hope it actually works ;)